core/src/main/scala/caliban/wrappers/ApolloTracing.scala

```diff
@@ -0,0 +1,187 @@
+package caliban.wrappers
+
+import java.time.{ Instant, ZoneId }
+import java.time.format.DateTimeFormatter
+import java.util.concurrent.TimeUnit
+import caliban.{ GraphQL, Rendering, ResponseValue }
+import caliban.ResponseValue.{ ListValue, ObjectValue }
+import caliban.Value.{ IntValue, StringValue }
+import caliban.wrappers.Wrapper.{ FieldWrapper, OverallWrapper, ParsingWrapper, ValidationWrapper }
+import zio.clock.Clock
+import zio.clock
+import zio.duration.Duration
+import zio.{ FiberRef, UIO }
+import zquery.ZQuery
+
+object ApolloTracing {
+
+  /**
+   * Attaches to the given GraphQL API definition a wrapper that adds tracing information to every response
+   * following Apollo Tracing format: https://github.com/apollographql/apollo-tracing.
+   * @param api a GraphQL API definition
+   */
+  def apolloTracing[R](api: GraphQL[R]): UIO[GraphQL[Clock with R]] =
+    FiberRef
+      .make(Tracing())
```

Comment:
Not familiar with how `FiberRef` works, so this may be a dumb question. But why can't this be lifted into an effect?

Couldn't this be:

```scala
UIO.effectSuspendTotal(FiberRef.make(Tracing()))
```

or could make an effect wrapper for `Tracing` as well.

```scala
for {
  tracing <- Tracing.make
  ref        <- FiberRef.make(tracing)
} yield api.withWrapper(apolloTracingOverallWrapper(ref))
```

Reply:
About 1, `FiberRef.make` already returns an effect, no need to use `effectSuspendTotal`.
About 2, I wanted to avoid forcing users having to do that and simply do `@@ apolloTracing` or something similar.

I'm trying with 2 new wrapper types `CombinedWrapper` and `EffectfulWrapper`, so that this logic would be "hidden".

Reply:
apolloTracing would be built this way:
```scala
EffectfulWrapper(
  FiberRef
    .make(Tracing())
    .map(
      ref =>
        apolloTracingOverall(ref) |+|
          apolloTracingParsing(ref) |+|
          apolloTracingValidation(ref) |+|
          apolloTracingField(ref)
    )
)
```
and then we are able to do:
```scala
graphQL(...) @@
  maxFields(200) @@
  maxDepth(30) @@
  timeout(3 seconds) @@
  printSlowQueries(500 millis) @@
  apolloTracing   
```